### PR TITLE
GameDB: EA Sports BIO FolderMC Support

### PR DESCRIPTION
**Pull request by PyraTheKitsune**
EA Sports BIO was an early form of cross-game profile system used in EA Sports' 2004 sports lineup. It was not used again for 2005, but the idea is novel enough and interesting enough that I felt putting a patch in was a good idea.

Here's an article about it: http://hitthepass.com/the-ea-sports-bio-it-was-in-the-game/
And another one (mostly about NBA Live 2004): https://www.nba-live.com/ww-ea-sports-bio-nba-live-2004/

This only works for NTSC-U as it's the only one I've tested, but I am planning on testing the PAL version as well.
The interesting thing is the use of PSCD10088 (no dash) as a serial. AFAIK that seems to be an internal thing, and not an official serial. Would be interesting if a dev could add some insight.

### Description of Changes
Add memCardFilter to all related games.

### Rationale behind Changes
Allows EA Sports BIO to be used with Folder Memory Cards.

### Suggested Testing Steps
Start the games involved, and save an EA Sports Bio, then save and load in each game, and see if it all reads/writes correctly.

### CODE REVIEW

**Comment by refractionpcsx2** [comment]
periods at the end of the comments please. (And don't close the PR again, thanks)

**Comment by refractionpcsx2** [comment]
uhh, you changed the line endings, so it made 46k line changes.

**Comment by PyraTheKitsune** [comment]
> uhh, you changed the line endings, so it made 46k line changes.

_facepalm_

That was a config error in my (admittedly hastily setup) IDE. Fixed in latest commit. It's been a long time since I've even touched

File: bin/resources/GameIndex.yaml [hunk 1]
```diff
@@ -41058,6 +41085,9 @@ SLUS-20867:
 SLUS-20868:
   name: "MVP Baseball 2004"
   region: "NTSC-U"
+  memcardFilters:
+    - "PSCD10088" # Enables EA Sports BIO (2004-only Cross Game Profile System)
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:41089]: still needs a period

File: bin/resources/GameIndex.yaml [hunk 2]
```diff
@@ -41232,6 +41262,9 @@ SLUS-20906:
   name: "Fight Night 2004"
   region: "NTSC-U"
   compat: 5
+  memcardFilters:
+    - "PSCD10088" # Enables EA Sports BIO (2004-only Cross Game Profile System)
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:41266]: still needs a period

**Comment by PyraTheKitsune** [comment]
Periods added, hopefully it's all clear. Apologies, it's 4am here now and I really should be in bed...

File: bin/resources/GameIndex.yaml [hunk 3]
```diff
@@ -41058,6 +41085,9 @@ SLUS-20867:
 SLUS-20868:
   name: "MVP Baseball 2004"
   region: "NTSC-U"
+  memcardFilters:
+    - "PSCD10088" # Enables EA Sports .
```
Comment (refractionpcsx2) [inline bin/resources/GameIndex.yaml:41089]: You deleted the word BIO :P When you put it back in make sure you don't leave an extra space in there, then I think we're golden.

**Comment by PyraTheKitsune** [comment]
I am an absolute klutz... but considering it's been a good 2 1/2 years since I touched a line of code, it's probably just nerves. Uploading the fixed version now. (I don't even have Git installed right now, in my hubris I thought this would be a quicker fix than it was.)

**Comment by refractionpcsx2** [comment]
It's fine, it's Git, it can always be fixed :)

It looks good now.

**Pull request merged**